pkg/exemplar/interface.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exemplar
+
+import (
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+// TsExemplar is an exemplar alongside a timestamp.
+type TsExemplar struct {
+	Ts       int64
+	Exemplar Exemplar
+}
+
+// SeriesExemplar is a time series of exemplar values.
+type SeriesExemplar []TsExemplar
```

Comment:
Perhaps this should be a `struct` then?
```Go
type SeriesExemplar struct{
    Labels      labels.Labels
    TsExemplars []TsExemplar
}
```

In that way, the consumer can actually see what labels matched the labels in the query.